packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -257,6 +321,26 @@ class ChangeNotifier implements Listenable {
         ));
       }
     }
+
+    _notificationCallStackDepth--;
+
+    if (_notificationCallStackDepth == 0 && _reentrantlyRemovedListeners > 0) {
+      // We really remove the listeners when all notifications are done.
+      final int newLength = _count - _reentrantlyRemovedListeners;
+      final List<VoidCallback?> newListeners = List<VoidCallback?>.filled(newLength, null);
```

Comment:
Should this follow the same logic as _removeAt and only resize if we are below a certain number of listeners and just compact otherwise?

Reply:
I've aligned this algorithm with the other one but with a logic a bit different since we can remove multiple listeners at the same time here. I also added a test for this specific use case. Can you look at it to confirm I didn't introduce any regression?